### PR TITLE
One like

### PR DESCRIPTION
**Pull request by 1ugia**
Trello - https://trello.com/c/5oHoyFln/57-l-bug-stop-user-from-liking-a-post-more-than-once

describe Like do
it 'allows users to only like one comment' do
click_button 'SUPRISE'
expect(page).to have_content('1 like')
end

it 'tells you off for liking more than once' do
click_button 'SURPRISE'
click_button 'SURPRISE'
expect(page).to raise('You can only like once')
end

end
<3

### CODE REVIEW

File: app/assets/stylesheets/likes.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the likes controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (YinnyF) [inline app/assets/stylesheets/likes.scss]: Could delete this file

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (YinnyF) [inline spec/models/like_spec.rb]: can delete this file

File: spec/requests/likes_spec.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe "Likes", type: :request do
+  describe "GET /index" do
+    pending "add some examples (or delete) #{__FILE__}"
+  end
+end
```
Comment (YinnyF) [inline spec/requests/likes_spec.rb:5]: Can delete this file

File: spec/helpers/likes_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# describe LikesHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe LikesHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (YinnyF) [inline spec/helpers/likes_helper_spec.rb]: Could delete this file

File: db/migrate/20210729171245_create_likes.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[6.0]
+  def change
+    create_table :likes do |t|
+      t.references :post, null: false, foreign_key: true
+      t.references :user, null: false, foreign_key: true
+
+      t.timestamps
+    end
+  end
+end
```
Comment (YinnyF) [inline db/migrate/20210729171245_create_likes.rb:10]: exactly how I dreamed it 😍

File: app/helpers/likes_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module LikesHelper
+end
```
Comment (YinnyF) [inline app/helpers/likes_helper.rb:2]: Can prob delete this

File: app/controllers/likes_controller.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    end
+    redirect_to URI(request.referer).path
+  end
+
+  private
+  def find_post
+    @post = Post.find(params[:post_id])
+  end
+
+  def already_liked?
```
Comment (YinnyF) [inline app/controllers/likes_controller.rb:17]: Nice, I like how this was refactored into its own method. 
Perhaps also make it a private method

File: app/controllers/posts_controller.rb [hunk 8]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
 
-  before_action :set_post, only: [:like, :edit, :update, :destroy]
+  before_action :set_post, only: [ :edit, :update, :destroy] # :like
```
Comment (YinnyF) [inline app/controllers/posts_controller.rb]: Is the comment necessary to keep?

File: app/controllers/posts_controller.rb [hunk 9]
```diff
@@ -11,6 +11,9 @@ def show
   def create
     params.inspect
     @post = Post.create(post_params)
```
Comment (YinnyF) [inline app/controllers/posts_controller.rb]: is this necessary given line 14?

File: app/controllers/posts_controller.rb [hunk 10]
```diff
@@ -40,6 +37,10 @@ def destroy
     redirect_to posts_path
   end
 
+  # def as_json(options = {})
+  #   super(options.merge(include: [:user, comments: {include: :user}]))
+  # end
+
```
Comment (YinnyF) [inline app/controllers/posts_controller.rb]: Could prob delete this right?

File: config/routes.rb [hunk 11]
```diff
@@ -7,7 +7,8 @@
   delete 'logout' => 'sessions#destroy'
   resources :posts do
     get '/like', to: 'posts#like', on: :member
```
Comment (YinnyF) [inline config/routes.rb:9]: Do we still need this given line 11?

File: app/views/posts/index.html.erb [hunk 12]
```diff
@@ -42,16 +45,11 @@
                       <%= image_tag(post.image, style: 'width:80%') %>
                     </p>
                   <% end %>
-                  <small>
-                    <a href="/posts/<%= post.id %>/like">Like</a>
-                    <br>
-                    <%= pluralize(post.likes,'like')%>
-                  </small>
+                  <%= button_to 'SUPRISE', post_likes_path(post), method: :post %>
+                  <p><%= post.likes.count %> <%= (post.likes.count) == 1 ? 'Like' : 'Likes'%></p>
```
Comment (YinnyF) [inline app/views/posts/index.html.erb]: `<%= post.likes.count %> <%= (post.likes.count) == 1 ? 'Like' : 'Likes'%>`
could be refactored into
`pluralize(post.likes.count, 'like')`

File: app/views/posts/index.html.erb [hunk 13]
```diff
@@ -42,16 +45,11 @@
                       <%= image_tag(post.image, style: 'width:80%') %>
                     </p>
                   <% end %>
-                  <small>
-                    <a href="/posts/<%= post.id %>/like">Like</a>
-                    <br>
-                    <%= pluralize(post.likes,'like')%>
-                  </small>
+                  <%= button_to 'SUPRISE', post_likes_path(post), method: :post %>
```
Comment (YinnyF) [inline app/views/posts/index.html.erb]: Why does it say 'SURPRISE' 🤔

**Review by YinnyF — CHANGES_REQUESTED** [review]
Very good, relatively clean, could delete some extra files that aren't needed

File: app/controllers/likes_controller.rb (hunk 7 above)
Comment (1ugia) [reply app/controllers/likes_controller.rb:17]: I just tried to put it in Private but the web broke 🙈

File: app/controllers/posts_controller.rb (hunk 9 above)
Comment (1ugia) [reply app/controllers/posts_controller.rb]: what about line 13? are they both the same?

File: app/helpers/likes_helper.rb (hunk 6 above)
Comment (1ugia) [reply app/helpers/likes_helper.rb:2]: Code broke without this so will have to keep it :3

File: config/routes.rb (hunk 11 above)
Comment (1ugia) [reply config/routes.rb:9]: yap, things break without it

File: app/controllers/posts_controller.rb (hunk 9 above)
Comment (1ugia) [reply app/controllers/posts_controller.rb]: I've left line 14 and  got rid of 13 instead.

File: app/views/posts/index.html.erb (hunk 13 above)
Comment (1ugia) [reply app/views/posts/index.html.erb]: 😂💪👍🏻

**Comment by 1ugia** [comment]
Done and ready to be re-reviewed :3

**Review by frodri13 — APPROVED** [review]
Excellent code. May need to revisit line 9 of likes_controller.rb

**Pull request merged**